scipy/cluster/tests/test_hierarchy.py

```diff
@@ -1251,24 +1274,24 @@ def test_cut_tree(xp):
     X = np.random.randn(nobs, 4)
     X = xp.asarray(X)
     Z = scipy.cluster.hierarchy.ward(X)
-    cutree = cut_tree(Z)
+    cutree = xp.asarray(cut_tree(Z), dtype=xp.int64)
```

Comment:
CI caught that this was having an `int32` dtype on some platforms. It was `int64` for all namespaces on my machine.

Reply:
But doesn't that mean that there is a legitimate test failure? I don't think we should just silence it. It is OK to silence it like this if you also open an issue. I don't think you need to *fix* the issues that `xp_assert_close` finds, especially not in this PR, but let's not hide them.

Reply:
Okay, I didn't know that this was a problem. Since the previous assertions were fine with mixed dtypes, I assumed that we didn't care whether we got an ‘int32` or an `int64`, as long as the assertions passed.

I'm happy to open an issue, but I'll need somebody to tell me where we are expecting specific dtypes.

Reply:
> I'll need somebody to tell me where we are expecting specific dtypes.

I would expect standard type promotion rules to be followed where applicable. In other cases, such as if a function returns integer indices, one would need to look at the documentation. If it's not documented, then perhaps it's fine for the type to vary, but then I'd set `check_dtype=False` to be explicit about the intent of the test rather than forcing the strict test to pass.

Please give it a shot with these things in mind, and ask a question about a particular test if there is a question.